src/Renci.SshNet/Security/KeyHostAlgorithm.cs

```diff
@@ -92,7 +92,7 @@ public BigInteger[] Keys
                     for (var i = 0; i < _keys.Count; i++)
                     {
                         var key = _keys[i];
-                        keys[i] = key.ToBigInteger();
+                        keys[i] = BigInteger.SshFormatBignum2(key);
```

Comment:
Care to explain why this is necessary? Would this break other algorithms?

Reply:
If a Key comes with a leading "FF", the current ToBigInteger() would drop this "FF" and we have invalid data. That gets prevented by appending an "00" before the "FF".

Made a lot of testruns with DSA and RSA and had no Problems.

Reply:
Can you add a test that covers this?
Haven't had time to check if this is trivial.

Reply:
Basically it's this: https://github.com/openssh/openssh-portable/blob/master/sshbuf-getput-crypto.c#L147